### DISCUSSION TITLE
'automerge' does not seem to work with 'matchFileNames'

### DISCUSSION THREAD

**Discussion by jw-maynard** [renovatebot/renovate#27622] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

When we have the following shared config config to update our GitHub actions which we import using `extends`. Renovate does not automerge the PRs it creates:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "github>PlayQ-DevOps/ci-config"
  ],
  "rangeStrategy": "replace",
  "packageRules": [
    {
      "matchManagers": [
        "github-actions"
      ],
      "matchFileNames": [
        "\\.github/workflows/[^/]+\\.yml$"
      ],
      "automerge": true
    }
  ]
}
```

If we change the config to remove the 'matchFileNames' settings rule then auto merge starts working.
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "github>PlayQ-DevOps/ci-config"
  ],
  "rangeStrategy": "replace",
  "packageRules": [
    {
      "matchManagers": [
        "github-actions"
      ],
      "automerge": true
    }
  ]
}
```

We don't have a minimum reproduction repo yet, but might be able to create one. We're also not sure which log lines in the DEBUG logs are "relevant" but we can pull them with some guidance as to what we should be looking for.

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment, marked as answer]
matchFileNames is glob matching, not regex. Can also be an exact string match.
